backend/src/main/java/com/huellapositiva/domain/actions/UpdateProposalImageAction.java

```diff
@@ -0,0 +1,73 @@
+import com.huellapositiva.domain.model.valueobjects.EmailAddress;
+import com.huellapositiva.domain.model.valueobjects.ProposalStatus;
+import com.huellapositiva.domain.repository.ProposalRepository;
+import com.huellapositiva.domain.service.EmailCommunicationService;
+import com.huellapositiva.domain.service.ImageService;
+import com.huellapositiva.domain.service.RemoteStorageService;
+import com.huellapositiva.infrastructure.orm.entities.JpaContactPerson;
+import com.huellapositiva.infrastructure.orm.entities.JpaProposal;
+import com.huellapositiva.infrastructure.orm.repository.JpaContactPersonRepository;
+import com.huellapositiva.infrastructure.orm.repository.JpaProposalRepository;
+import lombok.AllArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import javax.persistence.EntityNotFoundException;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.AccessDeniedException;
+
+@Service
+@AllArgsConstructor
+public class UpdateProposalImageAction {
+
+    @Autowired
+    private JpaContactPersonRepository jpaContactPersonRepository;
+
+    @Autowired
+    private JpaProposalRepository jpaProposalRepository;
+
+    @Autowired
+    private ProposalRepository proposalRepository;
+
+    @Autowired
+    private EmailCommunicationService emailCommunicationService;
+
+    @Autowired
+    private JpaContactPersonRepository contactPersonRepository;
+
+    @Autowired
+    private ImageService imageService;
+
+    private final RemoteStorageService remoteStorageService;
+
+    public void execute(MultipartFile photo, String accountId, String proposalId) throws IOException {
+        imageService.validateProfileImage(photo);
+        JpaContactPerson accountContactPerson = jpaContactPersonRepository.findByAccountId(accountId).orElseThrow(EntityNotFoundException::new);
+        JpaProposal jpaProposal = jpaProposalRepository.findByNaturalId(proposalId).orElseThrow(EntityNotFoundException::new);
+        validateProposalUpdate(accountContactPerson,jpaProposal);
+
+        URL url = remoteStorageService.uploadProposalImage(photo, proposalId);
+        Proposal proposal = Proposal.parseJpa(jpaProposal);
+        proposal.setImage(url);
+        proposal.setStatus(ProposalStatus.REVIEW_PENDING);
+        proposalRepository.save(proposal);
+
+        emailCommunicationService.sendProposalImageUpdateEmail(EmailAddress.from(accountContactPerson.getCredential().getEmail()));
+    }
+
+    private void validateProposalUpdate(JpaContactPerson accountContactPerson, JpaProposal jpaProposal) throws AccessDeniedException {
+        JpaContactPerson proposalContactPerson = contactPersonRepository.findByEsalId(jpaProposal.getEsal().getId()).orElseThrow(EntityNotFoundException::new);
+        if(!accountContactPerson.getCredential().getEmail().equals(proposalContactPerson.getCredential().getEmail())){
```

Comment:
We cannot compare users using a mutable property such as the email address, which is subject to be changed by the when updating the profile.

The users are not really identified by the email address, they are identified by ..., surprise, surprise, ..., their IDs. You simply need to check that the accountId of the logged user matches the one you are interested into.

Reply:
You are right, now we use getCredential.getId to compare